src/constructor.jl

```diff
@@ -190,6 +190,10 @@ function construct_mapping(dicttype::Union{Type,Function}, constructor::Construc
             end
         end
         try
+            if haskey(mapping, key)
+                @error "Duplicate key detected in mapping" node key
+                strict_unique_keys && error("Duplicate key `$(key)` detected in mapping.")
```

Comment:
Duplicate error in this case is a bit odd - what about something like:

```suggestion
                strict_unique_keys ? error("Duplicate key `$(key)` detected in mapping.") : @error "Duplicate key detected in mapping" node key
```

Reply:
Main reason for always doing the `@error` was that all of that is part of the `try catch` and will then immediately end up in a new `throw` of  `ConstructorError` - but I guess, handling proper logging would be the responsibility of any "user" (as in: for end-users the exception is fine; devs of another package should catch that and properly log it anyways).

Reply:
Yeah, I don't love having this in try/catch in the first place, that's not very idomatic julia. But I see that's legacy, so I think this is the best incremental change to address the problem.